### DISCUSSION TITLE
Group name acquisition and no lazy loading problem

### DISCUSSION THREAD

**Discussion by ezeli** [vaexio/vaex#2099] [Q&A] [answered]
When opening the hdf5 file, you must specify the group name defined at the time of creation. How should I obtain this group name if I have forgotten it ?
Another question is when opening hdf5 files, how to load them into memory all at once without lazy loading ?
Thank you !

**Comment by JovanVeljanoski** [comment, marked as answer]
Hi!

> When opening the hdf5 file, you must specify the group name defined at the time of creation. How should I obtain this group name if I have forgotten it ?

You can think of HDF5 files as a sort of containers. They do not have a pre-defined structure or a standard.. So in vaex we have an expectation of where the data should be (location) and how it should be stored (columnar format). But i think the gist of your question goes beyond vaex.. probably search elsewhere online how to "browse" the structure of an hdf5 files. Maybe with `h5py` it can be done..

> Another question is when opening hdf5 files, how to load them into memory all at once without lazy loading ?

Calling `df = df.materialize()` immediately after calling `vaex.open()` will put everything in RAM.
